src/Icicle/Avalanche/Statement/Simp/Mutate.hs

```diff
@@ -111,8 +114,11 @@ livevarsStmt ss = case ss of
   Let _ x s
    -> livevarsExp x <> livevarsStmt s
 
-  -- loop counter is live at the beginning and inside the loop.
-  ForeachInts n _ end body
+  -- loop counter is live after it is assigned the initial value,
+  -- and live inside the loop.
+  While _ n end body
+   -> Set.insert n (livevarsExp end <> livevarsStmt body)
```

Comment:
so I think the `acc` is live here, because it's used in the while body. For this case:

```
init acc = some array value V
let local = put_immutable x acc
while (acc == something) {...}
```

I don't think we want the put_immutable to be changed into a put_mutable?